src/box/lua/schema.lua

```diff
@@ -2488,18 +2488,22 @@ end
 
 box.internal.check_select_opts = check_select_opts -- for net.box
 
-local check_select_safety_rl = log.internal.ratelimit.new()
-local function check_select_safety(index, key_is_nil, itype, limit, offset,
+local function is_select_dangerous(sid, key_is_nil, itype, limit, offset,
                                    fullscan)
-    local rl = check_select_safety_rl
-    local sid = index.space_id
     local point_iter = itype == box.index.EQ or itype == box.index.REQ
     local window = offset + limit
-    if sid >= 512 and (key_is_nil or not point_iter) and
-       (not fullscan and window > 1000) then
-        rl:log_crit("Potentially long select from space '%s' (%d)\n %s",
-                    box.space[sid].name, sid, debug.traceback())
-    end
+    return sid >= 512 and
+           (key_is_nil or not point_iter) and
+           (not fullscan and window > 1000)
+end
+
+box.internal.is_select_dangerous = is_select_dangerous -- for read views
+
+local is_select_dangerous_rl = log.internal.ratelimit.new()
+local function log_dangerous_select(space)
+    local rl = is_select_dangerous_rl
+    rl:log_crit("Potentially long select from space '%s' (%d)\n %s",
+                space.name, space.id, debug.traceback())
```

Comment:
Nit: The ratelimit variable should be called `log_dangerous_select_rl`, because it's used by `log_dangerous_select`, not `is_select_dangerous`. Also, the local variable isn't really required.

<details><summary>diff</summary>

```diff
diff --git a/src/box/lua/schema.lua b/src/box/lua/schema.lua
index 0037bf5715b7..4e071975cc2f 100644
--- a/src/box/lua/schema.lua
+++ b/src/box/lua/schema.lua
@@ -2499,11 +2499,11 @@ end
 
 box.internal.is_select_dangerous = is_select_dangerous -- for read views
 
-local is_select_dangerous_rl = log.internal.ratelimit.new()
+local log_dangerous_select_rl = log.internal.ratelimit.new()
 local function log_dangerous_select(space)
-    local rl = is_select_dangerous_rl
-    rl:log_crit("Potentially long select from space '%s' (%d)\n %s",
-                space.name, space.id, debug.traceback())
+    log_dangerous_select_rl:log_crit(
+        "Potentially long select from space '%s' (%d)\n %s",
+        space.name, space.id, debug.traceback())
 end
 
 base_index_mt.select_ffi = function(index, key, opts)
```

</details>

Also, may be rename the function to `log_long_select` and `is_select_long` to match the log message?

Reply:
> Also, the local variable isn't really required.

It is, this is a closure and we need to keep state between calls.